[MASTRA-3074] LlamaIndex Extractor

NikAiyer opened the pull request:
## Description

<!-- Provide a brief description of the changes in this PR -->
This PR removes llamaindex extractors in favor of custom Mastra extractors.

## Related Issue(s)

<!-- Link to the issue(s) this PR addresses, using hashtag notation: #123 -->

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code refactoring
- [ ] Performance improvement
- [ ] Test update

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation (if applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have generated a changeset for this PR

NikAiyer committed:
started on converting extractors

NikAiyer committed:
Merge branch 'main' into mastra-3074/llamaindex-extract

NikAiyer committed:
added other extractors

NikAiyer committed:
updated extractors to use llamaindex/core

NikAiyer committed:
updated extractors

NikAiyer committed:
started on tests

NikAiyer committed:
updated extractor tests

NikAiyer committed:
updated testing

visini commented:
Interesting 🤩 Hopefully this resolves #2861, which is quite annoying and clogging up our logs. Thank you!

NikAiyer committed:
Merge branch 'main' into mastra-3074/llamaindex-extract

NikAiyer committed:
update tests

NikAiyer committed:
Merge branch 'mastra-3074/llamaindex-extract' of https://github.com/mastra-ai/mastra into mastra-3074/llamaindex-extract

NikAiyer committed:
updated test timings

NikAiyer committed:
Merge branch 'main' into mastra-3074/llamaindex-extract

NikAiyer committed:
updated test timing for document

NikAiyer committed:
updated extract docs

NikAiyer committed:
remove @llamaindex/env package

NikAiyer committed:
switch extractors to use llamaindex and remove @llamaindex/core

NikAiyer commented:
I switched everything to use llamaindex for now, and removed the @llamaindex/core and @llamaindex/env packages. I think we can switch to @llamaindex/core later once we determine our use cases.

NikAiyer committed:
Merge branch 'main' into mastra-3074/llamaindex-extract

NikAiyer committed:
updated import for tests

NikAiyer committed:
Merge branch 'mastra-3074/llamaindex-extract' of https://github.com/mastra-ai/mastra into mastra-3074/llamaindex-extract

NikAiyer committed:
Merge branch 'main' into mastra-3074/llamaindex-extract

NikAiyer committed:
Merge branch 'main' into mastra-3074/llamaindex-extract

NikAiyer committed:
Merge branch 'main' into mastra-3074/llamaindex-extract

NikAiyer committed:
Merge branch 'main' into mastra-3074/llamaindex-extract

NikAiyer committed:
Create long-keys-lick.md

NikAiyer committed:
Merge branch 'main' into mastra-3074/llamaindex-extract

NikAiyer committed:
added grouping for title extraction

NikAiyer committed:
Merge branch 'mastra-3074/llamaindex-extract' of https://github.com/mastra-ai/mastra into mastra-3074/llamaindex-extract

NikAiyer committed:
Merge branch 'main' into mastra-3074/llamaindex-extract

smorimoto commented:
See https://github.com/mastra-ai/mastra/issues/3770
